pkg/cosign/verify.go

```diff
@@ -602,27 +606,31 @@ func verifySignatures(ctx context.Context, sigs oci.Signatures, h v1.Hash, co *C
 	signatures := make([]oci.Signature, len(sl))
 	bundlesVerified := make([]bool, len(sl))
 
-	var wg sync.WaitGroup
-
+	t := throttler.New(co.MaxWorkers, len(sl))
 	for i, sig := range sl {
-		wg.Add(1)
 		go func(sig oci.Signature, index int) {
-			defer wg.Done()
 			sig, err := static.Copy(sig)
 			if err != nil {
 				validationErrs[index] = err.Error()
+				t.Done(nil)
```

Comment:
Instead of using `validationErrs`, it seems like we can pass in the error directly here via `t.Done(err)` and then later we can see which errors were created similar to this [example](https://github.com/nozzle/throttler/blob/master/example_test.go#L66-L72)

Reply:
yes, i did not applied that to not change the logic you made, but i will do that